tests/providers/http/hooks/test_http.py

```diff
@@ -525,3 +588,111 @@ async def test_async_request_uses_connection_extra(self, aioresponse):
+
+        assert extra_options == {}
+        assert actual == {"bearer": "test"}
+
+    def test_process_extra_options_from_connection_when_proxies_is_defined(self):
+        extra_options = {}
+        conn = get_airflow_connection_with_extra(
+            extra={"bearer": "test", "proxies": {"http": "http://proxy:80", "https": "https://proxy:80"}}
+        )()
+
+        actual = HttpAsyncHook._process_extra_options_from_connection(conn=conn, extra_options=extra_options)
+
+        assert extra_options == {"proxy": {"http": "http://proxy:80", "https": "https://proxy:80"}}
+        assert actual == {"bearer": "test"}
+
+    def test_process_extra_options_from_connection_when_proxy_is_defined(self):
+        extra_options = {}
+        conn = get_airflow_connection_with_extra(
+            extra={"bearer": "test", "proxy": {"http": "http://proxy:80", "https": "https://proxy:80"}}
+        )()
+
+        actual = HttpAsyncHook._process_extra_options_from_connection(conn=conn, extra_options=extra_options)
+
+        assert extra_options == {"proxy": {"http": "http://proxy:80", "https": "https://proxy:80"}}
+        assert actual == {"bearer": "test"}
+
+    def test_process_extra_options_from_connection_when_verify_is_defined(self):
+        extra_options = {}
+        conn = get_airflow_connection_with_extra(extra={"bearer": "test", "verify": False})()
+
+        actual = HttpAsyncHook._process_extra_options_from_connection(conn=conn, extra_options=extra_options)
+
+        assert extra_options == {"verify_ssl": False}
+        assert actual == {"bearer": "test"}
+
+    def test_process_extra_options_from_connection_when_verify_ssl_is_defined(self):
+        extra_options = {}
+        conn = get_airflow_connection_with_extra(extra={"bearer": "test", "verify_ssl": False})()
+
+        actual = HttpAsyncHook._process_extra_options_from_connection(conn=conn, extra_options=extra_options)
+
+        assert extra_options == {"verify_ssl": False}
+        assert actual == {"bearer": "test"}
+
+    def test_process_extra_options_from_connection_when_allow_redirects_is_defined(self):
+        extra_options = {}
+        conn = get_airflow_connection_with_extra(extra={"bearer": "test", "allow_redirects": False})()
+
+        actual = HttpAsyncHook._process_extra_options_from_connection(conn=conn, extra_options=extra_options)
+
+        assert extra_options == {"allow_redirects": False}
+        assert actual == {"bearer": "test"}
+
+    def test_process_extra_options_from_connection_when_max_redirects_is_defined(self):
+        extra_options = {}
+        conn = get_airflow_connection_with_extra(extra={"bearer": "test", "max_redirects": 3})()
+
+        actual = HttpAsyncHook._process_extra_options_from_connection(conn=conn, extra_options=extra_options)
+
+        assert extra_options == {"max_redirects": 3}
+        assert actual == {"bearer": "test"}
```

Comment:
This all looks like it's testing the same thing with different params, can this perhaps just all be in one test?

Reply:
This has been fixed and refactored as one test now